pkg/asset/installconfig/aws/metadata.go

```diff
@@ -12,17 +12,27 @@ import (
 // does not need to be user-supplied (e.g. because it can be retrieved
 // from external APIs).
 type Metadata struct {
-	session *session.Session
+	session           *session.Session
+	availabilityZones []string
+	region            string
 	sync.Mutex
 }
 
+// NewMetadata initializes a new Metadata object.
+func NewMetadata(region string) *Metadata {
+	return &Metadata{region: region}
+}
+
 // Session holds an AWS session which can be used for AWS API calls
 // during asset generation.
-// GetMetadata loads metadata.
 func (m *Metadata) Session(ctx context.Context) (*session.Session, error) {
 	m.Lock()
 	defer m.Unlock()
 
+	return m.unlockedSession(ctx)
+}
+
+func (m *Metadata) unlockedSession(ctx context.Context) (*session.Session, error) {
```

Comment:
I think this `unlocked` is not sustainable as we'll have to create these for each Getter.

Reply:
> I think this `unlocked` is not sustainable as we'll have to create these for each Getter.

For each getter that we consume from another getter.  Currently there is only this one, and I don't see it growing all that fast.  And it's not a lot of overhead to write an `unlocked` getter, so even if there is only one external consumer I think it's worth defining an `unlocked` getter instead of having the caller do something like:

```go
session, err := metadata.Session(ctx)
if err != nil {
  return err
}

zones, err := metadata.AvailabilityZones(ctx, session)
if err != nil {
  return err
}
```

And obviously as the number of external consumers grows, the benefit of a centralized `unlocked` helper increases.

Reply:
> . Currently there is only this one, and I don't see it growing all that fast.

sure.